### PR TITLE
feat(scenes): add GSCN format and EngineState integration

### PR DESCRIPTION
**Pull request by Isthimius**
## Summary

- clean the runtime Scene JSON graph:
  - serialize the layer collection as `SceneLayers` instead of `_sceneLayers`
  - stop serializing `SceneLayer -> Scene` and `SceneLayerTile -> SceneLayer` back-references
  - reconstruct ownership references during deserialization
  - continue reading legacy `_sceneLayers` state
  - correct stale Scene/SceneLayer ValueBag XML comments
- add `Gondwana.Scenes.GSCN`, following the existing GTS definition-file pattern:
  - `SceneDefinition`, layer/tile/frame/profile definition DTOs
  - `SceneDefinitionSerializer` for .gscn files, streams, JSON, packed assets, and runtime conversion
  - `SceneDefinitionSource` / `SceneDefinitionSourceKind`
  - `SceneDefinitionValidator`
  - lightweight tilesheet frame references rather than embedded tilesheet object graphs
  - sparse tile definitions supported for hand-authored/tool-generated scenes
  - collision values materialized before `*ByFrame` flags so frame-following semantics win
  - `CollisionsEnabled` derived from `CollisionType`, not duplicated in GSCN
- add `AssetTypes.SceneDefinition` for packed .gscn entries
- integrate GSCN with EngineState using the same inline/external pattern as GTS:
  - inline `SceneDefinition` entries by default
  - `separateGscnFiles: true` writes sibling `<state>.scenes/*.gscn` files
  - EngineState stores relative `GscnPath` references when possible
  - external GSCN files use `SceneDefinitionSerializer`, so they remain clean standalone documents
  - preserve backward compatibility with older EngineState files containing raw Scene graphs
- preserve canonical sprite/layer relationships across GSCN EngineState restore:
  - new sprite state stores `SceneId` + `SceneLayerId`
  - sprites rebind after scenes are materialized
  - new saves no longer embed duplicate SceneLayer graphs beneath sprites
  - legacy serialized SceneLayer references remain readable
- fix the existing `Scene.Empty` static-initialization edge case so the null-object scene is never registered as runtime state
- document GSCN files and EngineState inline/external scene behavior in the wiki

## Validation

Tests cover:
- cleaned runtime Scene JSON and legacy compatibility
- GSCN runtime -> definition -> JSON -> runtime parity
- restored parent ownership references
- scene collision groups/profiles
- layer presentation/wrapping/origin metadata
- tile frame/collision/animator metadata
- frame-derived collision values when explicit GSCN values are omitted
- omission of redundant `CollisionsEnabled` from GSCN JSON
- sparse GSCN tile materialization
- GSCN provenance stamping and validation diagnostics
- EngineState inline GSCN scenes
- EngineState external GSCN references/files
- LoadFromFile for inline and external scenes
- MergeFromFile Scene-ID overwrite semantics for inline and external scenes
- legacy raw-scene EngineState loading
- sprite rebinding to the canonical GSCN-restored SceneLayer
- `Scene.Empty` registry invariant

Current-head CI passed:
- solution build
- Gondwana.Tests unit tests (502 passed)
- WinForms editor interaction tests
- Blazor presentation helper tests
- pack validation
- publish/staging validation
- artifact creation/upload

### CODE REVIEW

File: Gondwana/Scenes/Scene.cs [hunk 1]
```diff
@@ -32,9 +32,27 @@ namespace Gondwana.Scenes;
...
 public class Scene : IEnumerable<SceneLayer>, IDisposable
 {
-    [JsonProperty]
+    [JsonProperty("SceneLayers")]
     private readonly List<SceneLayer> _sceneLayers = [];
 
+    // Read legacy state files that used the private field name without
+    // continuing to emit that implementation detail in new JSON.
+    [JsonProperty("_sceneLayers")]
+    private List<SceneLayer>? LegacySceneLayers
```
Comment (Isthimius) [inline Gondwana/Scenes/Scene.cs]: @copilot no need to include this, practically. just kill it

Comment (Isthimius) [reply Gondwana/Scenes/Scene.cs]: @copilot  remove this method
